src/components/future-appointments.jsx

```diff
@@ -0,0 +1,63 @@
+import { format, parseISO } from 'date-fns';
+import { Card, CardContent, CardHeader, CardTitle } from '@/components/ui/card';
+import { CalendarIcon, ClockIcon, UserIcon, HospitalIcon } from 'lucide-react';
+import { Spinner } from '@/components/ui/spinner';
+
+export default function FutureAppointments({ appointments, loading }) {
+  if (loading) {
+    return (
+      <Spinner>
+        <div className="text-center text-gray-500">
+          Loading appointments, please wait...
+        </div>
+      </Spinner>
+    );
+  }
+
+  if (appointments.length === 0) {
+    return (
+      <div className="text-center text-gray-500 py-8">
+        No appointments found 😕.
+      </div>
```

Comment:
can you use colors such as foreground to avoid hardcoding? It can look of in dark/light mode